frontend/amundsen_application/static/js/pages/HomePage/index.tsx

```diff
@@ -27,13 +27,57 @@ export interface DispatchFromProps {
   searchReset: () => UpdateSearchStateReset;
 }
 
+export type Widget = {
+  // TODO: What file does the Widget type belong in?
+  name: string;
+  options: {
+    path: '/';
+  };
+};
+
+export type HomePageLayout = Widget[];
+
 export type HomePageProps = DispatchFromProps & RouteComponentProps<any>;
 
 export class HomePage extends React.Component<HomePageProps> {
   componentDidMount() {
     this.props.searchReset();
   }
 
+  getHomePageWidgets(layout: HomePageLayout): React.ReactNode[] {
+    /* TODO: Make 100% sure it returns the pre-existing layout absent a custom config,
+    to not break any OSS users' Amundsen implementations */
+
+    return [
+      <div className="home-element-container">
+        <SearchBar />
+      </div>,
+      <div className="filter-breadcrumb pull-right">
+        <Breadcrumb
+          direction="right"
+          path="/search"
+          text={SEARCH_BREADCRUMB_TEXT}
+        />
+      </div>,
+      <div className="home-element-container">
+        <BadgesListContainer shortBadgesList />
+      </div>,
+      <div className="home-element-container">
+        <TagsListContainer shortTagsList />
+      </div>,
+      <div className="home-element-container">
+        <MyBookmarks />
+      </div>,
+      <div className="home-element-container">
+        <PopularTables />
+      </div>,
+    ];
+  }
+
+  defaultLayout: HomePageLayout = [];
+
+  homePageWidgets = this.getHomePageWidgets(this.defaultLayout);
```

Comment:
We don't really want to use instance variables, even if we are using classes. 
We'll be moving out of classes as soon as possible, so the less we depend on them, the better.